.github/workflows/release.yml

```diff
@@ -12,6 +18,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.events.inputs.tag || github.ref }}
```

Comment:
ref is:
> 

  | The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release,  this is the release tag created. For other triggers, this is the branch  or tag ref that triggered the workflow run. This is only set if a  branch or tag is available for the event type. The ref given is  fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1.
-- | --


from: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context